modules/deploy/main.tf

```diff
@@ -60,7 +60,10 @@ while [[ $STATUS == "Created" || $STATUS == "InProgress" || $STATUS == "Pending"
         --deployment-id $ID \
         --output text \
         --query '[deploymentInfo.status]')
-    sleep 5
+
+    SLEEP_TIME=$(( $RANDOM % 10 ) + ${var.get_deployment_sleep_timer})
+    echo "Sleeping for: $SLEEP_TIME Seconds"
+    sleep $SLEEP_TIME
```

Comment:
Are you sure that `aws deploy get-deployment` is being throttled? Sounds strange because it is a read type of operation even if it is executed multiple times at once.

Reply:
Hi @antonbabenko,

Thanks for your review.

In a busy day when we make multiple AWS API calls we get something like:
`An error occurred (ThrottlingException) when calling the GetDeployment operation (reached max retries: 4): Rate exceeded`

I think the following link describes the error (even if it's a few years old): https://forums.aws.amazon.com/thread.jspa?messageID=892511

Reply:
Right. Let's drop the `get_deployment_sleep_timer` variable because it is not necessary, but just sleep for 5 to 10 seconds, like this:

```
SLEEP_TIME=$(( $RANDOM % 5 ) + 5)
```

Short and simpler.

Reply:
I prefer to parameterize the sleep timer, as even if you are shifting 50% of the traffic every minute, you don't need to check the progress every ~ 5 seconds.


Is it a deal breaker for you if we keep the `get_deployment_sleep_timer` and set the default to 5?

Reply:
It is not a deal-breaker for me but I was not familiar with the use-case you had, so trying to balance between code simplicity and required flexibility. Let's keep it as you proposed. :)